File: AtCoder/ABC/169/python/D_Div_Game.py
N = int(input())


def prime_factorize(n):
    res = []
    for i in range(2, int(n ** 0.5) + 1):
        cnt = 0
        while n % i == 0:
            n /= i
            cnt += 1
        res.append(cnt)
    if n > 1:
        res.append(1)
    return res


v = prime_factorize(N)
res = 0
for cnt in v:
    lb = 0
    ub = cnt + 1
    while ub - lb > 1:
        md = (ub + lb) // 2
        if md * (md + 1) // 2 <= cnt:
            lb = md
        else:
            ub = md
    res += lb
print(res)

File: AtCoder/ABC/169/cpp/D_Div_Game.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

ll N;

map<ll, ll> prime_factorize(ll n) {
  map<ll, ll> res;
  for (ll i = 2; i * i <= n; ++i) {
    while (n % i == 0) {
      n /= i;
      ++res[i];
    }
  }
  if (n > 1) {
    ++res[n];
  }
  return res;
}

void solve() {
  map<ll, ll> m = prime_factorize(N);
  ll res = 0;
  for (auto _m : m) {
    ll v = _m.first;
    ll cnt = _m.second;
    ll lb = 0;
    ll ub = cnt + 1;
    while (ub - lb > 1) {
      ll md = (lb + ub) / 2;
      if (md * (md + 1) / 2 <= cnt) {
        lb = md;
      } else {
        ub = md;
      }
    }
    res += lb;
  }
  printf("%lld\n", res);
}

int main() {
  cin >> N;
  solve();
}